docs/src/main/asciidoc/getting-started-reactive.adoc

```diff
@@ -0,0 +1,840 @@
+
+[source, java]
+----
+package org.acme.vertx;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.ext.web.client.WebClient;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/fruit-data")
+public class ResourceUsingWebClient {
+
+    @Inject
+    Vertx vertx;
+
+    private WebClient client;
+
+    @PostConstruct
+    void initialize() {
+        this.client = WebClient.create(vertx,
+                new WebClientOptions().setDefaultHost("fruityvice.com").setDefaultPort(443).setSsl(true)
+                        .setTrustAll(true));
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/{name}")
+    public Uni<JsonObject> getFruitData(@PathParam("name") String name) {
+        return client.get("/api/fruit/" + name)
+                .send()
+                .map(resp -> {
+                    if (resp.statusCode() == 200) {
+                        return resp.bodyAsJsonObject();
+                    } else {
+                        return new JsonObject()
+                                .put("code", resp.statusCode())
+                                .put("message", resp.bodyAsString());
+                    }
+                });
+    }
+
+}
+----
+
+There are 2 important points:
+
+1. The injected Vert.x instance has the `io.vertx.mutiny.core.Vertx` type which is the Mutiny variant of Vert.x;
+2. The Web Client is created from `io.vertx.mutiny.ext.web.client.WebClient`.
+
+The Mutiny version of the Vert.x APIs also offers:
+
+* `andAwait` methods such as `sendAndAwait`. `andAwait` indicates that the caller thread is blocked until the result is available.
+Be aware not to block the event loop / IO thread that way.
+* `andForget` methods such as `writeAndForget`. `andForget` is available for method returning a `Uni`.
+`andForget` indicates that you don't need the resulting `Uni` indicating the success or failure of the operation.
+However, remember that if you don't subscribe, the operation would not be triggered.
+`andForget` manages this for you and manage the subscription.
+* `toMulti` methods allowing to transform a Vert.x `ReadStream` into a `Multi`
+* `toBlockingIterable` / `toBlockingStream` methods allowing to transform a Vert.x `ReadStream` into a blocking iterable or blocking `java.util.Stream`
+
+=== Using RxJava or Reactor APIs
+
+Mutiny provides utilities to convert RxJava 2 and Project Reactor types to `Uni` and `Multi`.
+
+RxJava 2 converters are available in the following dependency:
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.smallrye.reactive</groupId>
+    <artifactId>mutiny-rxjava</artifactId>
+    <version>$VERSION</version>
```

Comment:
version should comes from Quarkus bom or we risk having non-coherent versions of mutiny

Reply:
Yeah, if it's in the BOM, we could skip the version.

Reply:
It's not in the BOM :-) But I can add them.

Reply:
Defined in the BOM, documentation updated.